app/controllers/api/databases.php

```diff
@@ -368,6 +369,65 @@ function updateAttribute(
     return $attribute;
 }
 
+/**
+ * Recursively processes a collection to get it's relationships
+ *
+ * Returns true if the user has access to the collection
+ */
+function getCollectionRelationships(Document $collection, bool $mode, Database $dbForProject, int $depth, array &$collections): bool
+{
+    // Don't include this collection if the user doesn't have access to it
+    if (!($mode === APP_MODE_ADMIN && Auth::isPrivilegedUser(Authorization::getRoles()))) {
+        if (!$collection->getAttribute('documentSecurity', false)) {
+            $validator = new Authorization(Database::PERMISSION_READ);
+            if (!$validator->isValid($collection->getRead())) {
+                return false;
+            }
+        }
+    }
+
+    if ($depth + 1 > Database::RELATION_MAX_DEPTH) {
+        return true;
+    }
+
+    $relationshipAttributes = \array_filter(
+        $collection->getAttribute('attributes', []),
+        fn ($attribute) => $attribute->getAttribute('type') === Database::VAR_RELATIONSHIP
+    );
+
+    $attributes = [];
+    foreach ($relationshipAttributes as $attribute) {
+        /** @var Document $attribute */
+        $relatedCollection = Authorization::skip(fn() => $dbForProject->getDocument('database_' . $collection->getAttribute('databaseInternalId'), $attribute->getAttribute('relatedCollection')));
```

Comment:
Instead of skipping auth, can we utilize the permission check at the database level and check if `$relatedCollection->isEmpty()`, then skip the recursion if it is? Then we can remove the manual check at the start of this function too